crates/cheatnet/tests/common/mod.rs

```diff
@@ -103,8 +110,31 @@ pub fn call_contract(
         initial_gas: u64::MAX,
     };
 
+    let mut execution_resources = ExecutionResources::default();
+    let mut entry_point_execution_context = EntryPointExecutionContext::new(
+        &build_block_context(runtime_state.cheatnet_state.block_info),
+        &build_transaction_context(),
+        ExecutionMode::Execute,
+        false,
+    )
+    .unwrap();
+    let hints = HashMap::new();
+
+    let mut syscall_hint_processor = SyscallHintProcessor::new(
+        blockifier_state.blockifier_state,
+        &mut execution_resources,
+        &mut entry_point_execution_context,
+        Relocatable {
```

Comment:
Why have you selected this pointer?

Reply:
This pointer doesn't do anything here, syscall handler is passed there only to share mut refs to resources, context and state. I can write a comment explaining that

Reply:
I also think that this solution is not ideal, maybe another struct passing only the 3 things we actually use would be good, but idk if it wouldn't just introduce a bunch of unnecessary mess